code/game/objects/items.dm

```diff
@@ -127,15 +127,23 @@
 	return ..(user, distance, "", message)
 
 /obj/item/attack_hand(mob/user as mob)
-	var/atom/old_loc = src.loc
-
-	src.pickup(user)
+	src.before_pickup(user)
```

Comment:
This doesn't fit with established design patterns, look at pre_equip for an example

It should be called pre_pickup, and it should return a value which dictates whether or not the pickup action completes.

something like
if (!pre_pickup(user))
return